### PR TITLE
Implementation of a new design of the Bottom Bar

### PR DESCRIPTION
**Pull request by didi2101**
# Added Features:

- **color.kt**  
- **BottomBar.kt**  
- **BottomBarTest.kt**

# Summary:

This PR introduces a redesigned BottomBar with improved visuals, new buttons, and comprehensive test coverage. These changes enhance user experience through a modernized interface, dynamic animations, and robust navigation functionality.

### LIST_OF_TOP_LEVEL_DESTINATIONS:
Adds new screens to `LIST_OF_TOP_LEVEL_DESTINATIONS` to integrate them seamlessly with the updated BottomBar. Ensures proper navigation and alignment with the overall app structure.

### color.kt:
Introduces a new blue gradient for the BottomBar background, contributing to a refreshed and polished aesthetic. Includes updated color definitions for consistency across the app.

### BottomBar.kt:
Implements a completely new BottomBar design featuring:
- Animated transitions for improved interactivity.
- Highlighted states for selected screens.

### BottomBarTest.kt:
Adds a new test class to validate the functionality and design of the BottomBar. The tests include:
- Check that all screen buttons are displayed correctly.
- Validation of highlighted states for selected screens and unselected screens.
- Ensures proper navigation actions for each destination within the BottomBar.

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/composables/BottomBar.kt [hunk 1]
```diff
@@ -1,48 +1,102 @@
...
-              horizontalAlignment = Alignment.CenterHorizontally) {
-                IconButton(
-                    modifier = Modifier.testTag(destination.textId),
-                    onClick = { navigationActions.navigateTo(destination.route) }) {
-                      Icon(
-                          imageVector = destination.icon,
-                          contentDescription = destination.textId,
-                          tint = Color.White,
-                          modifier = Modifier.size(30.dp))
+  navigationActions.currentRoute()
```
Comment (nadaedh) [inline app/src/main/java/com/android/sample/ui/composables/BottomBar.kt:39]: what's the point of calling this here ?

**Review by nadaedh — COMMENTED** [review]
The new BottomBar design is an excellent addition to the app and works as intended.  Great job!

File: app/src/main/java/com/android/sample/ui/composables/BottomBar.kt (hunk 1 above)
Comment (didi2101) [reply app/src/main/java/com/android/sample/ui/composables/BottomBar.kt:39]: You are right.
Fixed

File: app/src/main/java/com/android/sample/ui/settings/SettingsScreen.kt [hunk 2]
```diff
@@ -48,6 +49,7 @@ fun SettingsScreen(
 
   Scaffold(
       modifier = Modifier.testTag("settingsScreen"),
+      bottomBar = { BottomBar(navigationActions = navigationActions) },
```
Comment (Siupahc) [inline app/src/main/java/com/android/sample/ui/settings/SettingsScreen.kt:52]: Good fix

File: app/src/main/java/com/android/sample/ui/workout/SessionSelectionScreen.kt [hunk 3]
```diff
@@ -35,6 +36,7 @@ import com.android.sample.ui.theme.White
 fun SessionSelectionScreen(navigationActions: NavigationActions) {
   Scaffold(
       topBar = { TopBar(navigationActions, R.string.NewWorkout) },
+      bottomBar = { BottomBar(navigationActions = navigationActions) },
```
Comment (Siupahc) [inline app/src/main/java/com/android/sample/ui/workout/SessionSelectionScreen.kt:39]: Here too

File: app/src/main/java/com/android/sample/ui/composables/TopBar.kt [hunk 4]
```diff
@@ -32,7 +31,7 @@ import com.android.sample.ui.theme.White
 fun TopBar(navigationActions: NavigationActions, @StringRes title: Int) {
 
   androidx.compose.foundation.layout.Box(
-      modifier = Modifier.fillMaxWidth().wrapContentHeight().background(brush = TopBarGradient)) {
+      modifier = Modifier.fillMaxWidth().wrapContentHeight().background(brush = BlueGradient)) {
```
Comment (Siupahc) [inline app/src/main/java/com/android/sample/ui/composables/TopBar.kt:34]: Good standardisation

**Review by Siupahc — APPROVED** [review]
Good fixes, lgtm

**Pull request merged**